Update security considerations section and update text for clarity (#43)
danwing/cidfi: draft-wing-cidfi.md

Indication message allows the CIDFI network element to distinguish
each QUIC client's UDP 4-tuple.

> The default size of a short Destination Connection ID (DCID) is 8 bytes, 

Reviewer: Please cite that section in the text. Thanks

~~~~~
{: #fig-import artwork-align="left" title="Example JSON for Flow Importance"}

> Note: {{fig-import}} lists sample attributes and they will be discussed in detail in a separate document.

Reviewer: ```suggestion
> Note: {{fig-import}} lists sample attributes and they will be discussed in detail in a separate document.
```

client knows the identity of the server and can validate
its certificate.

> Spoofing Attack: For an attacker to succeed with the nonce challenge against a victim's UDP 4-tuple, the attacker has to send a STUN CIDFI-NONCE packet using the victim's source IP address and a valid HMAC. A valid HMAC can only be obtained by the attacker making its own connection to the CIDFI server and spoofing the source IP address and UDP port of the victim. If the client does not support CIDFI, the attacker can influence the packet treatment of the victim's 5-tuple. If the client implements CIDFI, the CIDFI network element can identify an IP address spoofing attack . The CIDFI network element receives two HTTPS connections describing the same DCID (one connection from the attacker, another from the victim). The CIDFI network element will issue unique Nonces and HMACs to both the attacker and victim, and both the attacker and victim will send the STUN indication on that same UDP 4-tuple. It should generate an alarm on the CIDFI network element. In this scenario, it is recommended that both the attacker and the victim be denied CIDFI access. Such spoofing of a victim's IP address is prevented by the network using network ingress filtering ({{!RFC2827}}, {{!RFC7513}}, {{!RFC6105}}, and/or {{!RFC6620}}).

Reviewer: ```suggestion
Spoofing Attacks:
: For an attacker to succeed with the nonce challenge against a victim's UDP 4-tuple, an attacker has to send a STUN CIDFI-NONCE packet using the victim's source IP address and a valid HMAC. A valid HMAC can only be obtained by the attacker making its own connection to the CIDFI-aware server and spoofing the source IP address and UDP port number of the victim.
: If the client does not support CIDFI, the attacker can influence the packet treatment of the victim's UDP 4-tuple.
: If the client implements CIDFI, a CIDFI network element can identify an IP address spoofing attack. Concretely, the CNE will receive two HTTPS connections describing the same DCID; one connection from the attacker and another one from the victim. The CNE will then issue unique Nonces and HMACs to both the attacker and victim, and both the attacker and victim should send the STUN Indication on that same UDP 4-tuple. Such an event should trigger an alarm on the CNE. In this scenario, it is recommended that both the attacker and the victim be denied CIDFI access.
: The spoofing of a victim's IP address is prevented by the network using network ingress filtering ({{!RFC2827}}, {{!RFC7513}}, {{!RFC6105}}, and/or {{!RFC6620}}).
```

> On-Path Attack: An on-path attacker can observe the victim's Discovery Packet, block it, and then forward the packet within the attacker's 5-tuple. Subsequently, the on-path attacker can 'steal' the victim's CIDFI control from the victim's 5-tuple, causing the victim's CIDFI signaling for that 5-tuple to influence the attacker's 5-tuple. Although the on-path attacker can't directly observe the encrypted CIDFI signaling, this attack effectively disables the victim's CIDFI treatment, making it accessible to the attacker. The attacker can send NEW_CONNECTION_ID frames to the server with the victim's (observed) Destination CID, effectively claiming the victim's CIDFI signaling for themselves. To mitigate this replay attack, shared networks require per-endpoint encryption (e.g., Wi-Fi WPA3, DOCSIS BPI+) or traffic separation (e.g., Ethernet switching rather than bridging).

Reviewer: ```suggestion
On-Path Attacks:
: An on-path attacker can observe the victim's Discovery Packet, block it, and then forward the packet within the attacker's 5-tuple. Subsequently, the on-path attacker can 'steal' the victim's CIDFI control from the victim's UDP 4-tuple, causing the victim's CIDFI signaling for that UDP 4-tuple to influence the attacker's UDP 4-tuple.
: Although the on-path attacker can't directly observe the encrypted CIDFI signaling, this attack effectively disables the victim's CIDFI treatment, making it accessible to the attacker. The attacker can send NEW_CONNECTION_ID frames to the server with the victim's (observed) Destination CID, effectively claiming the victim's CIDFI signaling for themselves. To mitigate this replay attack, shared networks require per-endpoint encryption (e.g., Wi-Fi WPA3, DOCSIS BPI+) or traffic separation (e.g., Ethernet switching rather than bridging).
```

> On-Path Attack: An on-path attacker can observe the victim's Discovery Packet, block it, and then forward the packet within the attacker's 5-tuple. Subsequently, the on-path attacker can 'steal' the victim's CIDFI control from the victim's 5-tuple, causing the victim's CIDFI signaling for that 5-tuple to influence the attacker's 5-tuple. Although the on-path attacker can't directly observe the encrypted CIDFI signaling, this attack effectively disables the victim's CIDFI treatment, making it accessible to the attacker. The attacker can send NEW_CONNECTION_ID frames to the server with the victim's (observed) Destination CID, effectively claiming the victim's CIDFI signaling for themselves. To mitigate this replay attack, shared networks require per-endpoint encryption (e.g., Wi-Fi WPA3, DOCSIS BPI+) or traffic separation (e.g., Ethernet switching rather than bridging).

Reviewer: > The attacker can send NEW_CONNECTION_ID frames to the server with the..

But this will be rejected by the server.

Author: A QUIC client is free to use use any DCID; multiple QUIC clients within the the same endpoint itself can use the same DCID.

Reviewer: My comment was about an attacker who sends a NEW_CONNECTION_ID. Adding or retiring CIDs is encrypted and thus don't see on which case an attacker can inject those and pass through the validation at the server side.

Author: The on-path attacker can create a new QUIC connection using the same DCID.

Reviewer: But if creates a new connection, then the initial DCID will be set to zero. 

Also, how this interferes with the connection that was established by the client.

Author: The on-path attacker creates a new QUIC connection with the same DCID as the victim to server, it drops the STUN indication message from client and sends STUN indication message to prove ownership of the 5-tuple. It means, the victim will not be able to prioritize packets. The connection from the victim will continue to exit.

Author: The relay attack is not applicable with a on-path attacker who is capable of dropping the traffic from victim. I will remove it.

Reviewer: Thanks for removing that part. 

Still need a figure with the various messages to digest the attack.

Author: [cidfi-attack.pptx](https://github.com/danwing/cidfi/files/13437765/cidfi-attack.pptx)
Attached figure explaining the on-path attack.

Reviewer: Thanks. The file points to the generic cidfi prez.

Author: [cidfi-attack.pptx](https://github.com/danwing/cidfi/files/13438288/cidfi-attack.pptx)

Author: Please see the updated presentation.
